mrcnn/model.py

```diff
@@ -3099,6 +3100,35 @@ def unmold_outputs(detections, mrcnn_class, mrcnn_bbox, mrcnn_mask,
     scores: [N] float probability scores for the class IDs
     masks: [H, W, N] instance binary masks
     """
+
+    if len(args) == 8:
+        (
+            detections,
+            mrcnn_class,
+            mrcnn_bbox,
+            mrcnn_mask,
+            rpn_rois,
+            rpn_class,
+            rpn_bbox,
+            image_metas
+        ) = args
+        image_class = None
+    elif len(args) == 9:
+        (
+            detections,
+            mrcnn_class,
+            mrcnn_bbox,
+            mrcnn_mask,
+            rpn_rois,
+            rpn_class,
+            rpn_bbox,
+            image_metas,
+            image_class
+        ) = args
```

Comment:
You always need to set `image_class`, otherwise it will be undefined below.
```suggestion
    if len(args) in [8, 9]:
        (detections, mrcnn_class, mrcnn_bbox, mrcnn_mask,
         rpn_rois, rpn_class, rpn_bbox, image_metas, 
         image_class) = (args + [None])[:9]
```

Reply:
I'm setting it. But your code is shorter.

Reply:
Sry, wood from the trees...